pyro/infer/mcmc/hmc.py

```diff
@@ -93,6 +108,15 @@ def _get_trace(self, z):
         trace_poutine(*self._args, **self._kwargs)
         return trace_poutine.trace
 
+    @staticmethod
+    def _iter_latent_nodes(trace):
+        for name, node in sorted(trace.iter_stochastic_nodes(), key=lambda x: x[0]):
```

Comment:
Why sort by name rather than the default order (by execution time)?

Reply:
Yeah, any fixed order is fine. The advantage to using names is that the dict based data structures that we are using can be sorted anytime by key names (otherwise, we'll need to store the key order somewhere and then sort using that order).